crates/sui-core/src/consensus_adapter.rs

```diff
@@ -361,60 +523,43 @@ impl ConsensusAdapter {
                 warn!("Still waiting {total_wait} seconds for transaction {transaction_key:?} to commit in narwhal");
             }
         }));
-        if let Some(processed_waiter) = processed_waiter {
-            debug!("Submitting {transaction_key:?} to consensus");
 
-            // populate the position only when this authority submits the transaction
-            // to consensus
-            guard.position = Some(position);
+        // later populate this with the narwhal nodes that had lowest reputation scores
+        let exclusions = vec![];
 
-            // We enter this branch when in select above await_submit completed and processed_waiter is pending
-            // This means it is time for us to submit transaction to consensus
-            {
-                let ack_start = Instant::now();
-                let mut retries: u32 = 0;
-                while let Err(e) = self
-                    .consensus_client
-                    .submit_to_consensus(&transaction, epoch_store)
-                    .await
-                {
-                    // This can happen during Narwhal reconfig, so wait for a few retries.
-                    if retries > 3 {
-                        error!(
-                            "Error submitting transaction to own narwhal worker: {:?}",
-                            e
-                        );
-                    }
-                    self.opt_metrics.as_ref().map(|metrics| {
-                        metrics.sequencing_certificate_failures.inc();
-                    });
-                    retries += 1;
-                    time::sleep(Duration::from_secs(10)).await;
-                }
+        let should_submit = self.should_submit(
+            epoch_store.committee(),
+            &self.authority,
+            &transaction,
+            exclusions.clone(),
+        );
 
-                // we want to record the num of retries when reporting latency but to avoid label
-                // cardinality we do some simple bucketing to give us a good enough idea of how
-                // many retries happened associated with the latency.
-                let bucket = match retries {
-                    0..=10 => retries.to_string(), // just report the retry count as is
-                    11..=20 => "between_10_and_20".to_string(),
-                    21..=50 => "between_20_and_50".to_string(),
-                    51..=100 => "between_50_and_100".to_string(),
-                    _ => "over_100".to_string(),
-                };
+        if should_submit {
+            self.submit_after_selected(transaction.clone(), epoch_store)
+                .await;
+        }
 
+        // We wait to ensure the transaction was received by consensus. If sufficient time passes
+        // and consensus hasn't received the transaction, we submit it ourselves as a fallback.
+        let sleep_timer = tokio::time::sleep(Duration::from_secs(7));
```

Comment:
Can we try to retain as much of existing logic as possible with the change?

* We can keep the order and `wait_submit` logic
* Limit to only removing the disconnected peers from the list before calculating our own position